multiraft/multiraft.go

```diff
@@ -869,6 +876,21 @@ func (s *state) handleMessage(req *RaftMessageRequest) {
 		}
 	}
 
+	// Ignore all message execept heartbeat when recevie snapshot until apply finish.
+	g := s.groups[req.GroupID]
+	g.pendingSnapshotLock.Lock()
+	if g.pendingSnapshot {
+		g.pendingSnapshotLock.Unlock()
+		return
+	}
+
+	if req.Message.Type == raftpb.MsgSnap {
+		if req.Message.Snapshot.Metadata.Index != 10 {
```

Comment:
Why is the initial snapshot handled differently?

We shouldn't hard-code the number 10 here. If we need to we should move `raftInitialLogIndex` from `storage` to `multiraft` and export it.